src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -494,15 +494,12 @@ public void TestMainModule()
         {
             Process p = Process.GetCurrentProcess();
 
-            // On UAP casing may not match - we use Path.GetFileName(exePath) instead of kernel32!GetModuleFileNameEx which is not available on UAP
-            Func<string, string> normalize = PlatformDetection.IsInAppContainer ?
-                (Func<string, string>)((s) => s.ToLowerInvariant()) :
-                (s) => s;
-
             Assert.InRange(p.Modules.Count, 1, int.MaxValue);
-            Assert.Equal(normalize(RemoteExecutor.HostRunnerName), normalize(p.MainModule.ModuleName));
-            Assert.EndsWith(normalize(RemoteExecutor.HostRunnerName), normalize(p.MainModule.FileName));
-            Assert.Equal(normalize(string.Format("System.Diagnostics.ProcessModule ({0})", RemoteExecutor.HostRunnerName)), normalize(p.MainModule.ToString()));
+            Assert.Equal(RemoteExecutor.HostRunnerName, p.MainModule.ModuleName);
+            Assert.EndsWith(RemoteExecutor.HostRunnerName, p.MainModule.FileName);
+            Assert.Equal(string.Format("System.Diagnostics.ProcessModule ({0})", RemoteExecutor.HostRunnerName), p.MainModule.ToString());
+
+            Assert.Equal(Environment.ProcessPath, p.MainModule.FileName);
```

Comment:
If it's not too difficult, it would be nice to have a test asserting that `Environment.ProcessPath` will not change during program execution even if you rename the executable, seeing as we want to guarantee that. Admittedly, I'm not sure how difficult that would be with the way we run tests, so maybe it's not worth the trouble.

Reply:
This is not easy in the current test harness. The tests are executed using `dotnet test.dll` today. I would need to rename `dotnet` that would cause all sorts of the problems, e.g. any attempts to start tests would fail while `dotnet` is renamed.

I can add tests to validate that the property is returning identical string instance. It will verify that the caching works.